### PR TITLE
[NAN-697] load a script from a directory that is the provider config

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Allow nested directories to be used in a nango-integrations directory. There will be follow ups to update the `init` command and also add a migrate command.

Allows for
```
nango-integrations/
├── nango.yaml               
└── hubspot/
    ├── syncs/
    │   └── contacts.ts      
    └── actions/
        └── create-contact.ts
```

and is backwards compatible. 

## Issue ticket number and link
NAN-697

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [x] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

**Review by TBonnin — APPROVED** [review]

File: packages/cli/lib/sync.unit.test.ts [hunk 1]
```diff
@@ -451,4 +469,40 @@ describe('generate function tests', () => {
...
+
+        const success = await compileService.run({ debug: true });
+
+        await fs.promises.rm('./hubspot', { recursive: true, force: true });
+        await fs.promises.rm('./github', { recursive: true, force: true });
+        await fs.promises.rm('./dist', { recursive: true, force: true });
+        await fs.promises.rm('./nango.yaml', { force: true });
+        await fs.promises.rm('./models.ts', { force: true });
+
+        expect(success).toBe(true);
```
Comment (bodinsamuel) [inline packages/cli/lib/sync.unit.test.ts:488]: there is no verification that nested file have been indeed compiled no?

**Review by bodinsamuel — APPROVED** [review]
Working locally 👍🏻 
Might be good to actually document this alternative somewhere

File: packages/cli/lib/sync.unit.test.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/cli/lib/sync.unit.test.ts:488]: No, but if not compiled successfully, success won't be true.

**Comment by khaliqgant** [comment]
A follow up to this will be to allow a sync/action name to be duplicated within different providers.
```
hubspot:
   syncs:
     contacts:

salesforce:
  syncs:
    contacts:
```

**Comment by khaliqgant** [comment]
> Working locally 👍🏻 Might be good to actually document this alternative somewhere

Yeah there is a specific documentation task along with a few others for the migration command etc

**Pull request merged**